lib/src/adapters/persistent_json_api.dart

```diff
@@ -144,8 +144,13 @@ class PersistentJsonApiAdapter extends JsonApiAdapter {
     Iterable<String> ids,
   ) async {
     var box = await openBox(endpoint);
-    var docs =
-        ids.map((id) => box.get(id)).where((doc) => doc != null).toList();
+    List<JsonApiDocument> docs = [];
```

Comment:
@algodave instead of imperatively filling the list we can probably just add `.cast<String>()` to make `docs` List generic type non-nullable String.

`var docs = ids.map((id) => box.get(id)).where((doc) => doc != null).toList().cast<String>();`